### ISSUE TITLE
QA Environment for Pull Requests

### ISSUE THREAD

**Issue by swissspidy** [GoogleForCreators/web-stories-wp#1398]
We kinda touched on this a little bit in #1170, but it merits its own ticket.

**tl;dr**: We want to make it easier for QA to test PRs before they land in master. Various options exist, and we need to define what our next steps are.

**What we have right now**

- Every push to `master` is automatically deployed to our testing environement
- Every pull request automatically has a ZIP file attached to it that can be downloaded and installed on a WordPress site.\*

\* To get the ZIP file:

1. Open pull request
2. Click on "Checks" at the top  
    <img width="668" alt="Screenshot 2020-04-30 at 11 05 49" src="https://user-images.githubusercontent.com/841956/80692983-a6894280-8ad2-11ea-8c5c-2c30ce89dc5f.png">
3. Click on "Continuous Integration" on the left-hand side  
    <img width="656" alt="Screenshot 2020-04-30 at 11 06 08" src="https://user-images.githubusercontent.com/841956/80693011-b0ab4100-8ad2-11ea-80aa-a1167c25fd14.png">
4. Download `web-stories.zip`  
    <img width="799" alt="Screenshot 2020-04-30 at 11 06 20" src="https://user-images.githubusercontent.com/841956/80693026-b6a12200-8ad2-11ea-8ae0-76dc279f3250.png">

**Possible options**

1. Do nothing  
   We decide that using the provided ZIP files is enough for QA testing, and nothing further needs to be done.
1. Introduce QA Tester WordPress plugin  
   Similar to what Site Kit is doing, this would be a simple WordPress plugin that one installs on their site. This plugin allows you to easily select a PR you want to test, and then it downloads and installs the ZIP file mentioned above.  
   Downside: this is not much faster than just downloading the ZIP file yourself.
1. Automatically spin up a test environment for every PR (or every PR with a specific label)  
1. Have an environment similiar to http://gutenberg.run/ -- type in a PR number and have a fresh test environment spun up for you.

**Comment by swissspidy** [comment]
cc @pbakaus @kmyram @csossi for initial thoughts. Can bring this up in a weekly meeting at some point (perhaps after stable when we have more time).

**Comment by swissspidy** [comment]
gutenberg.run source code, just for reference: https://github.com/aduth/gutenberg.run/

**Comment by swissspidy** [comment]
Example from the QA Tester plugin:

<img width="594" alt="Screenshot 2020-05-06 at 16 29 22" src="https://user-images.githubusercontent.com/841956/81189411-cfb84000-8fb6-11ea-80db-adddebb06ef2.png">
